### DISCUSSION TITLE
getServerSideProps is not called when re-entering page

### DISCUSSION THREAD

**Discussion by stefan-girlich** [vercel/next.js#45629] [Help] [answered]
### Scenario
- I am using `getServerSideProps` individually on two pages which are linked to each other via `router.push()` and via `<Link />`
- Next.js v12.2.5
- Issue appears only in production builds, never when using dev server
- I check if `getServerSideProps` is executed via `console.log`

### Actual behavior

1. Open page A => `getServerSideProps` A called
2. Navigate to page B (via `<Link/>`) => `getServerSideProps` B called
3. Navigate back to page A (via `router.push()`) => `getServerSideProps` A _not_ called
4. Navigate again to page B => `getServerSideProps` B _not_ called

### Expected behavior
`getServerSideProps` is called reliably when a page is entered via link or router (see [docs](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#when-does-getserversideprops-run))

I first suspected that the response headers for the Next.js JSON data could cause caching, but the only cache header explicitly disables caching (`Cache-Control: private, no-cache, no-store, max-age=0, must-revalidate`).

Would anyone know what the issue could be? Thanks so much!!

**Comment by icyJoseph** [comment, marked as answer]
A couple of things to clear up:

## middleware

> I'd check this one first!

Do you have it activated? I seem to recall that there was a bug were GSSP was being pre-fetched when middleware existed, and I seem to recall it was fixed in `12.2.6`

- https://github.com/vercel/next.js/issues/38189#issuecomment-1173846297
- https://github.com/vercel/next.js/issues/38918#issuecomment-1227213946

You can check if this is the issue by looking at the network tab in dev tools, when is the GSSP `.json` call made? when you render the link (that'd be prefetch), on hover? (prefetch too), or is it made when you actually click on the link?

## Service Worker

If the above doesn't apply, then are you using something like `next-pwa`? It uses a service worker on https://developer.chrome.com/docs/workbox/modules/workbox-strategies/#stale-while-revalidate, which prevents gSSP from being fetched on demand, rather a cache'd version is served.

**Reply by stefan-girlich** [reply]
Thanks for the extensive and super-quick response!! 
I have updated Next.js to v12.3.4 which fixed the issue :)

**Reply by icyJoseph** [reply]
Awesome! Yeah I do recall there was a few version with that subtle bug... Please don't forget to mark as answered!
